### PR TITLE
added private learning with multiple teachers

### PR DESCRIPTION
**Pull request by npapernot**
This code, as describe in the README file, trains private models on MNIST, CIFAR10, and SVHN by using noisy knowledge transfer and aggregation.

### CODE REVIEW

File: privacy/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Knowledge acquired by teachers is transferred to the student in a differentially
+private manner by noisily aggregating the teacher decisions before feeding them
+to the student during training.
+
+A paper describing the approach is in preparation. A link will be added to this 
+README when available.
+
+## Dependencies
+
+This model uses `tensorflow` to perform numerical computations associated with
```
Comment (benoitsteiner) [inline privacy/README.md]: Can you spell TensorFlow with an upper case T and an upper case F?

File: privacy/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+For instance, to train teacher number 10 among an ensemble of 100 teachers for 
+MNIST, you use the following command:
+
+```
+python train_teachers.py --nb_teachers=100 --teacher_id=10 --dataset=mnist
+```
+
+Other flags like `train_dir` and `data_dir` should optionally be set to
+respectively point to the directory where model checkpoints and temporary data
+(like the dataset) should be save. The flag `max_steps` (default at 3000)
```
Comment (benoitsteiner) [inline privacy/README.md]: s/save/saved/

File: privacy/deep_cnn.py [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+    weights = _variable_with_weight_decay('weights', 
+                                          shape=[384, 192],
+                                          stddev=0.04, 
+                                          wd=0.004)
+    biases = _variable_on_cpu('biases', [192], tf.constant_initializer(0.1))
+    local4 = tf.nn.relu(tf.matmul(local3, weights) + biases, name=scope.name)
+    if dropout:
+      local4 = tf.nn.dropout(local4, 0.5, seed=FLAGS.dropout_seed)
+
+  # softmax, i.e. softmax(WX + b)
```
Comment (benoitsteiner) [inline privacy/deep_cnn.py]: Can you update this comment, since there is no softmax here.

File: privacy/deep_cnn.py [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+    weights = _variable_with_weight_decay('weights',
+                                          shape=[192, 192],
+                                          stddev=0.05,
+                                          wd=0)
+    biases = _variable_on_cpu('biases', [192], tf.constant_initializer(0.1))
+    local2 = tf.nn.relu(tf.matmul(local1, weights) + biases, name=scope.name)
+    if dropout:
+      local2 = tf.nn.dropout(local2, 0.5, seed=FLAGS.dropout_seed)
+
+  # softmax, i.e. softmax(WX + b)
```
Comment (benoitsteiner) [inline privacy/deep_cnn.py]: Can you also update this comment since there's no actual softmax here?

**Comment by npapernot** [comment]
I signed it.
